### PR TITLE
add nonlinear test

### PR DESCRIPTION
**Pull request by fserra**
this merge request translates the tests string, circle and gasnet from SCIP to PySCIPOpt

it also allows to set `lb = None` when creating a variable for having -infinite lower bound

### CODE REVIEW

**Comment by fserra** [comment]
I am guessing this is timing out because there is no ipopt

File: src/pyscipopt/expr.pxi [hunk 1]
```diff
@@ -530,7 +530,6 @@ cdef class GenExpr:
         return otherexpr.__div__(self)
 
     def __truediv__(self,other):
-        print("truediv of GenExpr", self, other)
```
Comment (mattmilten) [inline src/pyscipopt/expr.pxi]: good one! :-)

File: src/pyscipopt/scip.pyx [hunk 2]
```diff
@@ -740,6 +740,8 @@ cdef class Model:
         cname = str_conversion(name)
         if ub is None:
             ub = SCIPinfinity(self._scip)
+        if lb is None:
+            lb = -SCIPinfinity(self._scip)
```
Comment (mattmilten) [inline src/pyscipopt/scip.pyx]: weird! I don't know why this was missing...

**Comment by mattmilten** [comment]
The time outs are annoying...

**Comment by mattmilten** [comment]
Maybe adding  a time limit might be a solution to produce a valid status for the test.

**Comment by fserra** [comment]
don't get it, how would this timelimit work? I guess the best would be able to have ipopt, either by having an installer that contains it or compile it there?

**Comment by mattmilten** [comment]
We could check for the right result or status time limit. This would
effectively skip the test. Sure, in a perfect world we would always have
Ipopt available...

**Pull request merged**